sklearn/utils/estimator_checks.py

```diff
@@ -1635,13 +1662,13 @@ def check_non_transformer_estimators_n_iter(name, Estimator):
 
     # LassoLars stops early for the default alpha=1.0 the iris dataset.
     if name == 'LassoLars':
-        estimator = Estimator(alpha=0.)
```

Comment:
(not the fault of this PR but) Just curious why can't we check if n_iter_ is at least 0 instead. LassoLars is doing the right thing: on some datasets there is no work to be done.